src/components/AddSupplementalPosition/AddSupplementalPosition.tsx

```diff
@@ -1,121 +1,151 @@
-        onClose && onClose();
-    };
+export const AddSupplementalPosition = ({ readOnly }: AddSupplementalPositionProps) => {
+    const {
+        register,
+        setValue,
+        control,
+        trigger,
+        watch,
+        formState: { errors },
+    } = useFormContext<SupplementalPositionsType>();
 
-    const onOpen = () => {
-        formVisibility.setTrue();
-        setPercentage(0.01);
-    };
+    const { fields, append, remove } = useFieldArray({
+        control,
+        name: 'supplementalPositions',
+        rules: { maxLength: 1 },
+    });
 
     return (
         <div className={s.Base}>
-            {nullable(
-                formVisibility.value,
-                () => (
-                    <div>
-                        <div className={s.Header}>
-                            <Text as="h3">{tr('Supplemental position')}</Text>
+            <div>
+                {fields.map((field, index) => {
+                    const workStartDate = watch(`supplementalPositions.${index}.workStartDate`);
+                    return (
+                        <div key={field.id}>
+                            <div className={s.Header}>
+                                <Text as="h3">{tr('Supplemental position')}</Text>
 
-                            {nullable(!readOnly, () => (
-                                <AddInlineTrigger
-                                    text={tr('Remove supplemental position')}
-                                    icon={<IconBinOutline size="s" />}
-                                    onClick={onReset}
-                                />
-                            ))}
+                                {nullable(!readOnly, () => (
+                                    <AddInlineTrigger
+                                        text={tr('Remove supplemental position')}
+                                        icon={<IconBinOutline size="s" />}
+                                        onClick={() => remove(index)}
+                                    />
+                                ))}
+                            </div>
+                            <div className={s.TwoInputsRow}>
+                                <FormControl label={tr('Supplemental organization')} required>
+                                    <OrganizationUnitComboBox
+                                        readOnly={readOnly}
+                                        searchType="internal"
+                                        onChange={(orgUnit) =>
+                                            orgUnit &&
+                                            setValue(`supplementalPositions.${index}.organizationUnitId`, orgUnit.id)
+                                        }
+                                        organizationUnitId={watch(`supplementalPositions.${index}.organizationUnitId`)}
+                                        error={
+                                            errors?.supplementalPositions instanceof Array &&
```

Comment:
Why `supplimentalPositions` cannot be Array? If this field can be `null` or `Array` then i think will be better make non strict compare with `null`

Also may be create an additional errors handler which get errors object, index of current group fields and prop ex.

```ts
const errorPicker = (errors, i, key) => {
  if (error == null) return;
  return errors[index][key]
}
```

```ts
errorPicker(errors.supplementalPositions, index, 'organizationUnitId')
```

Its makes your code more clear

Reply:
errorPicker great idea, thanks!